pkg/urbit/noun/serial.c

```diff
@@ -856,6 +864,443 @@ u3s_cue_atom(u3_atom a)
   return u3s_cue_bytes((c3_d)len_w, byt_y);
 }
 
+/* _cs_etch_ud_size(): output length in @ud for given mpz_t.
+*/
+static inline size_t
+_cs_etch_ud_size(mpz_t a_mp)
+{
+  size_t len_i = mpz_sizeinbase(a_mp, 10);
+  return len_i + (len_i / 3); // separators
+}
+
+/* _cs_etch_ud_bytes(): atom to @ud impl.
+*/
+static size_t
+_cs_etch_ud_bytes(mpz_t a_mp, size_t len_i, c3_y* hun_y)
+{
+  c3_y*   buf_y = hun_y + (len_i - 1);
+  mpz_t   b_mp;
+  c3_w     b_w;
+  size_t dif_i;
+
+  mpz_init2(b_mp, 10);
+
+  if ( !mpz_size(a_mp) ) {
+    *buf_y-- = '0';
+  }
+  else {
+    while ( 1 ) {
+      b_w = mpz_tdiv_qr_ui(a_mp, b_mp, a_mp, 1000);
+      c3_assert( mpz_get_ui(b_mp) == b_w ); // XX
+
+      if ( !mpz_size(a_mp) ) {
+        while ( b_w ) {
+          *buf_y-- = '0' + (b_w % 10);
+          b_w /= 10;
+        }
+        break;
+      }
+
+      *buf_y-- = '0' + (b_w % 10);
+      b_w /= 10;
+      *buf_y-- = '0' + (b_w % 10);
+      b_w /= 10;
+      *buf_y-- = '0' + (b_w % 10);
+      *buf_y-- = '.';
+    }
+  }
+
+  buf_y++;
+
+  c3_assert( buf_y >= hun_y ); // XX
+
+  //  mpz_sizeinbase may overestimate by 1
+  //
+  {
+    size_t dif_i = buf_y - hun_y;
+
+    if ( dif_i ) {
+      len_i -= dif_i;
+      memmove(hun_y, buf_y, len_i);
+      memset(hun_y + len_i, 0, dif_i);
+    }
+  }
+
+  mpz_clear(b_mp);
+
+  return len_i;
+}
+
+/* u3s_etch_ud_smol(): c3_d to @ud
+**
+**   =(26 (met 3 (scot %ud (dec (bex 64)))))
+*/
+c3_y*
+u3s_etch_ud_smol(c3_d a_d, c3_y hun_y[26])
+{
+  c3_y*  buf_y = hun_y + 25;
```

Comment:
`26` is used multiple times in this implementation, a large enough number of times to warrant a `static const` variable or `#define` at the top of the file.